Airwallex/Core/Sources/Internal/Models/AWXCard.swift

```diff
@@ -0,0 +1,169 @@
+        case brand
+        case country
+        case funding
+        case fingerprint
+        case cvcCheck = "cvc_check"
+        case avsCheck = "avs_check"
+        case numberType = "number_type"
+    }
+
+    public init(number: String? = nil, expiryMonth: String?, expiryYear: String?, name: String?, cvc: String?, bin: String?, last4: String?, brand: String?, country: String?, funding: String?, fingerprint: String?, cvcCheck: String?, avsCheck: String?, numberType: String?) {
+        self.number = number
+        self.expiryMonth = expiryMonth
+        self.expiryYear = expiryYear
+        self.name = name
+        self.cvc = cvc
+        self.bin = bin
+        self.last4 = last4
+        self.brand = brand
+        self.country = country
+        self.funding = funding
+        self.fingerprint = fingerprint
+        self.cvcCheck = cvcCheck
+        self.avsCheck = avsCheck
+        self.numberType = numberType
+    }
+
+    public required init(from decoder: any Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        number = try container.decodeIfPresent(String.self, forKey: .number)
+        expiryMonth = try container.decodeIfPresent(String.self, forKey: .expiryMonth)
+        expiryYear = try container.decodeIfPresent(String.self, forKey: .expiryYear)
+        name = try container.decodeIfPresent(String.self, forKey: .name)
+        cvc = try container.decodeIfPresent(String.self, forKey: .cvc)
+        bin = try container.decodeIfPresent(String.self, forKey: .bin)
+        last4 = try container.decodeIfPresent(String.self, forKey: .last4)
+        brand = try container.decodeIfPresent(String.self, forKey: .brand)
+        country = try container.decodeIfPresent(String.self, forKey: .country)
+        funding = try container.decodeIfPresent(String.self, forKey: .funding)
+        fingerprint = try container.decodeIfPresent(String.self, forKey: .fingerprint)
+        cvcCheck = try container.decodeIfPresent(String.self, forKey: .cvcCheck)
+        avsCheck = try container.decodeIfPresent(String.self, forKey: .avsCheck)
+        numberType = try container.decodeIfPresent(String.self, forKey: .numberType)
+
+        if number == nil, let last4 = last4 {
+            number = "••••\(last4)"
+        }
+    }
+}
+
+@objc public extension AWXCard {
+    func validateAndReturnError() -> String? {
+        if number == nil || !AWXCardValidator.shared.isValidCardLength(number ?? "") {
+            return "Invalid card number"
+        }
+        if name == nil || name?.count == 0 {
+            return "Invalid name on card"
+        }
+        if expiryYear == nil || expiryMonth == nil || expiryYear?.count == 0 || expiryMonth?.count == 0 {
+            return "Invalid expires date"
+        }
+        if cvc == nil || cvc?.count == 0 {
+            return "Invalid CVC / CVV"
+        }
+        return nil
+    }
+
+    static func decodeFromJSON(_ dic: [String: Any]) -> AWXCard {
```

Comment:
Seems this is only used by tests, and the only other response model that has AWXCard is AWXPaymentMethod, which is also swift, and nowhere else will call it from OC? 🤔

Reply:
Oh just noticed you mentioned it here https://github.com/airwallex/airwallex-payment-ios/pull/80#discussion_r1718222519, so we can apply same strategy here.